pkg/proxy/iptables/proxier.go

```diff
@@ -1619,7 +1619,7 @@ func (proxier *Proxier) syncProxyRules() {
 
 	// Close old local ports and save new ones.
 	for k, v := range proxier.portsMap {
-		if replacementPortsMap[k] == nil {
+		if replacementPortsMap[k] == nil && v != nil {
```

Comment:
is this related to the PR?

Reply:
I don't think so. With or without the PR, we could encounter panic here if we specify `nodePort` in unit tests, so I think we could handle it in the defensive way.

Reply:
:thinking: 
https://stackoverflow.com/a/42239079/7794348
> 
In Go, however, the function to be called by the Expression.Name() syntax is entirely determined by the type of Expression and not by the particular run-time value of that expression, including nil. In this manner, the invocation of a method on a nil pointer of a specific type has a clear and logical meaning. Those familiar with vtable[] implementations will find this odd at first, yet, when thinking of methods this way, it is even simpler and makes sense. Think of:

Reply:
Well I think the root cause was [here](https://github.com/kubernetes/kubernetes/blob/v1.21.0-alpha.2/pkg/proxy/iptables/proxier_test.go#L464-L467):
```golang
// OpenLocalPort fakes out the listen() and bind() used by syncProxyRules
// to lock a local port.
func (f *fakePortOpener) OpenLocalPort(lp *utilproxy.LocalPort, isIPv6 bool) (utilproxy.Closeable, error) {
	f.openPorts = append(f.openPorts, lp)
	return nil, nil
}
```
The `fakePortOpener` in our test is returning `nil` for both `Closeable` and `error`, where our code made an assumption that if `error` is `nil`, then we should be able to close the `Closeable`. I can revert that change and just add a fake `Closeable` instead of returning nil to keep that assumption.

Reply:
cool, you solved 2 issues in one
Thanks